rust/agama-lib/src/storage/client.rs

```diff
@@ -178,20 +81,20 @@ impl<'a> StorageClient<'a> {
 
     pub async fn volume_for(&self, mount_path: &str) -> Result<Volume, ServiceError> {
         let volume_hash = self.calculator_proxy.default_volume(mount_path).await?;
-        let volume = Volume {
-            mount_path: get_property(&volume_hash, "MountPath")?,
-            mount_options: get_property(&volume_hash, "MountOptions")?,
-            target: get_property(&volume_hash, "Target")?,
-            target_device: get_optional_property(&volume_hash, "TargetDevice")?,
-            min_size: get_property(&volume_hash, "MinSize")?,
-            max_size: get_optional_property(&volume_hash, "MaxSize")?,
-            auto_size: get_property(&volume_hash, "AutoSize")?,
-            snapshots: get_optional_property(&volume_hash, "Snapshots")?,
-            transactional: get_optional_property(&volume_hash, "Transactional")?,
-            outline: get_optional_property(&volume_hash, "Outline")?,
-        };
-
-        Ok(volume)
+
+        Ok(volume_hash.try_into()?)
+    }
+
+    pub async fn product_mount_points(&self) -> Result<Vec<String>, ServiceError> {
```

Comment:
Add to your personal TODO list to document these methods :-) Actually, I expect some refactoring (because we might need an HTTP-based client) so it can be part of that change.